src/Core/Addon/Theme/ThemeManager.php

```diff
@@ -352,14 +360,49 @@ private function installFromZip($source)
         $sandboxPath = $this->getSandboxPath();
         Tools::ZipExtract($source, $sandboxPath);
```

Comment:
Legacy class in Core? 🤔

Reply:
not my part :laughing:  I only added more exceptions here . I think this code part is from the very beginning of the PrestaShop 1.7 - if I gonna remove legacy here then it might be a BC break I think

Reply:
not necessary, depends if you change the public API or not

Reply:
anyway, I think ia huge refactoring here is needed but I am not sure if it should be targeted to 1.7.6.x. Right now it calls command `ImportThemeCommand` which calls `ImportThemeHandler` and inside it calls existing  ThemeManager which has dependencies on legacy not only in this place. 

My suggestion here would be to handle everything in ImportThemeHandler and don't even touch the `ThemeManager` but I think its a lot of work and it would be hard to fit in the 1.7.6.x scope